sdk/resourcemanager/Azure.ResourceManager/src/ArmAudience.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager
+{
+    /// <summary>
+    /// ArmAudience represents the audience of Azure Cloud.
+    /// </summary>
+    public class ArmAudience : IEquatable<ArmAudience>
+    {
+        // name after the `name` property of returned audience from https://management.azure.com/metadata/endpoints?api-version=2019-11-01
+        /// <summary> Azure Public Cloud. </summary>
+        public static readonly ArmAudience AzureCloud = new ArmAudience("https://management.azure.com/");
```

Comment:
Does this now double as the Uri as well?  Should we remove those params from the ArmClient ctor and use this from the options?

Reply:
Hmm... this is a good point. If the audience is always matches the base URI provided when the client is constructed the `Audience` option might not be needed at all. We've only been adding this option for service clients where we can't determine the audience from the service endpoint provided in client construction. For instance services which support custom domain names. If this isn't the case for ARM we might be able to remove the option all together.

Reply:
There are two issues we need to consider.

1. From the [response of this API](https://management.azure.com/metadata/endpoints?api-version=2019-11-01), there could be multiple audience per endpoint. Will the audience grow? If so, then we cannot use its value as the base URI.
2. Audience has [a trailing slash issue](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-permissions-and-consent#trailing-slash-and-default). For audience, we need to keep the extra slash, but for base URI we should not keep it.

Reply:
Take Java SDK as reference. They bundle the environment information altogether in [AzureEnvironment](https://github.com/Azure/azure-sdk-for-java/blob/2cd30e97440042b17a94db3438865ed530fb4c08/sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java).

- `managementEndpointUrl` is actually audience. I get confirmation that they never use it as a URL.
- `resourceManagerEndpointUrl` is the mgmt endpoint URL.

Besides, they allow users to specify their own scope in https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/arm/AzureConfigurable.java#L80

Reply:
Lets follow the Java pattern to keep customers from being able to specify incompatible values.